### PR TITLE
fix: support http token-ID inputs for v1/completions and v1/embeddings

### PR DESCRIPTION
**Pull request by zetxqx**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

This PR implements support for pre-tokenized inputs (token IDs) in `/v1/completions` and `/v1/embeddings` endpoints. Previously, these endpoints did not accept token arrays, leading to parse failures or incorrect token count estimations (silent length corruption).

The OpenAI API spec allows `prompt` and `input` to be arrays of integers (token IDs). Our EPP request-body parser only supported strings or arrays of strings. This caused parse errors when clients sent pre-tokenized inputs, or resulted in an estimated token count of 0.

## Changes
1.  Extended `Prompt` and created `EmbeddingsInput` to support `[]int` (TokenIDs).
2.  Added `InputTokenCountHint()` to provide exact token counts for token-ID inputs, avoiding inaccurate string-based estimation.
3.  Optimized `UnmarshalJSON` in `types.go` to decode once into `any` and perform type inspection, improving performance.
7.  Updated `vllmgrpc` parser to correctly populate `TokenIDs` from gRPC `TokenizedInput`, enforcing oneof semantics by setting only one field in `Prompt`/`EmbeddingsInput`.

**Which issue(s) this PR fixes**:

Fixes partially #2835

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/epp/framework/plugins/requestcontrol/requestdataproducer/predictedlatency/plugin.go [hunk 1]
```diff
@@ -280,10 +279,17 @@ func newPredictedLatencyContext(request *framework.InferenceRequest) *predictedL
...
+	if request.Body != nil {
+		inputTokenCount = request.Body.InputTokenCountHint()
+	}
+	if inputTokenCount < 0 {
+		inputTokenCount = len(strings.Fields(promptText))
+	}
 	return &predictedLatencyCtx{
 		schedulingRequest:             *request,
 		promptText:                    promptText,
-		inputTokenCount:               len(strings.Fields(promptText)),
```
Comment (kaushikmitr) [inline pkg/epp/framework/plugins/requestcontrol/requestdataproducer/predictedlatency/plugin.go:286]: there are few other places where we are using len(strings.Fields(prompt)) or len(strings.Fields(promptText)), in this plugin. Can you consolidate all of them to just use inputTokenCount?

File: pkg/epp/framework/plugins/requestcontrol/requestdataproducer/predictedlatency/plugin.go
Comment (kaushikmitr) [inline pkg/epp/framework/plugins/requestcontrol/requestdataproducer/predictedlatency/plugin.go:278]: May be add a Unit test showing newPredictedLatencyContext produces a non-zero inputTokenCount for a token-ID prompt and for a string embeddings input." Not present.

File: pkg/epp/framework/interface/requesthandling/types.go [hunk 2]
```diff
@@ -130,18 +145,74 @@ func (r *InferenceRequestBody) CacheSalt() string {
...
+		}
+		return arrayInputResult{Strings: strings}, nil
+	case float64:
+		ints := make([]int, len(v))
+		for i, val := range v {
+			flt, ok := val.(float64)
+			if !ok {
+				return arrayInputResult{}, fmt.Errorf("%s: mixed types in array", errorPrefix)
+			}
+			ints[i] = int(flt)
```
Comment (kaushikmitr) [inline pkg/epp/framework/interface/requesthandling/types.go]: does this incorrecly accepts inputs like: [1.5, 2.7].

Comment (zetxqx) [reply pkg/epp/framework/interface/requesthandling/types.go]: The JSON unmarshal will treat the numbers as float64 by default. Added a check to make sure this is integer. Also I changed the int to unint32. in bd5c857c65c27dd3426959373a44770ff88384c2.

File: pkg/epp/framework/plugins/requestcontrol/requestdataproducer/predictedlatency/plugin.go
Comment (zetxqx) [reply pkg/epp/framework/plugins/requestcontrol/requestdataproducer/predictedlatency/plugin.go:278]: I ended up having too many changes. And some changes I'm not exactly sure I'm changing it right. So I removed the changes in latency predictor. We can do a follow up in llm-d to migrate latency predictor to use the new tokenCountHint. What do you think?

File: pkg/epp/framework/plugins/requestcontrol/requestdataproducer/predictedlatency/plugin.go (hunk 1 above)
Comment (zetxqx) [reply pkg/epp/framework/plugins/requestcontrol/requestdataproducer/predictedlatency/plugin.go:286]: I removed the changes in latency predictor. We can do a follow up in llm-d-inference-scheduler. I can scope this PR to parser only
